resume-buddy/src/main/java/com/google/sps/Match.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.util.List;
+
+/** Class that matches reviewers to reviewees */
+public class Match {
+  List<Entity> reviewees;
+  List<Entity> reviewers;
+
+  public Match() {
+    getNotMatchedUsers();
+    match();
+  }
+
+  /** Retrieves reviewers' and reviewees' data from the Datastore */
+  private void getNotMatchedUsers() {
+    Query revieweeQuery = new Query("Reviewee");
+    Query reviewerQuery = new Query("Reviewer");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery revieweeResults = datastore.prepare(revieweeQuery);
+    PreparedQuery reviewerResults = datastore.prepare(reviewerQuery);
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(Integer.MAX_VALUE);
+
+    reviewees = revieweeResults.asList(fetchOptions);
+    reviewers = reviewerResults.asList(fetchOptions);
+  }
+
+  /** FCFS algorithm to match reviewers with reviewees */
+  private void match() {
+    // TODO: Update algorithm based on criteria 
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    int numberOfMatched = Math.min(reviewees.size(), reviewers.size());
+
+    for (int i = 0; i < numberOfMatched; i++) {
+      Entity reviewer = reviewers.get(i);
+      Entity reviewee = reviewees.get(i);
```

Comment:
Is it guaranteed that you'll get the reviewers and reviewees in order of which they submitted their signup forms? 

If not, you may have to store the timestamp of when each reviewer and reviewee submitted the signup form. Then query and sort the timestamps (see https://cloud.google.com/datastore/docs/concepts/queries#sort_orders).

Reply:
I am almost sure that it works so. I tested by hand and it matches users in order, also the webpage says "If no sort orders are specified, the results are returned in the order they are retrieved from Datastore". I'' add timestamps to be completely certain. 

Adding timestamps is also good idea for complex algorithm, for instance, if we have 2 or more perfect matches we should match with the one who filled out form first.

Reply:
Yeah, I think timestamps would be good for later on. You may want an expiration period, for example, so you don't match someone who entered the form > 6 months ago or something like that. Feel free to add a TODO to add a timestamp or do it in this PR.